controllers/configurationpolicy_controller.go

```diff
@@ -351,10 +385,36 @@ func (r *ConfigurationPolicyReconciler) handleObjectTemplates(
 				return
 			}
 
-			if templates.HasTemplate(objectT.ObjectDefinition.Raw, "") {
+			if templates.HasTemplate(objectT.ObjectDefinition.Raw, "", true) {
 				log.V(1).Info("Processing policy templates")
 
 				resolvedTemplate, tplErr := tmplResolver.ResolveTemplate(objectT.ObjectDefinition.Raw, nil)
+
+				// If the error is because the padding is invalid, this either means the encrypted value was not
+				// generated by the "protect" template function or the AES key is incorrect. Control for a stale
+				// cached key.
+				if usedKeyCache && errors.Is(tplErr, templates.ErrInvalidPKCS7Padding) {
+					log.V(2).Info(
+						"The template decryption failed likely due to an invalid encryption key, will refresh " +
+							"the encryption key cache and try the decryption again",
+					)
+					var encryptionConfig templates.EncryptionConfig
+					encryptionConfig, usedKeyCache, tplErr = r.getEncryptionConfig(plc, true)
+
+					if tplErr == nil {
+						tmplResolverCfg.EncryptionConfig = encryptionConfig
+
+						tmplResolver, err = templates.NewResolver(&kubeclient, config, tmplResolverCfg)
+						if err != nil {
+							// Panic here since this error is unrecoverable
+							log.Error(err, "Failed to create a template resolver")
```

Comment:
@mprahl  : why is it panic here ? would I want all other non template policies to be affected by this, couldnt this be just an error on that policy itself ?